jobdescription.py

```diff
@@ -0,0 +1,517 @@
+                    'scope': scope[i],
+                    "guid": guids[i]
+                }
+    return files
+
+
+def fix_log(description, files):
+    """
+    Fixes log file description in output files (changes GUID and scope).
+
+    :param description:
+    :param files: output files
+    :return: fixed output files
+    """
+    log.info("modifying log-specific values in a log file description")
+    if description["logFile"] and description["logFile"] != "NULL":
+        if description["logGUID"] and description["logGUID"] != "NULL" and description["logFile"] in \
+                files:
+            files[description["logFile"]]["guid"] = description["logGUID"]
+            files[description["logFile"]]["scope"] = description["scopeLog"]
+
+    return files
+
+
+def get_output_files(description):
+    """
+    Extracts output files from the description.
+
+    :param description:
+    :return: output files
+    """
+    log.info("Extracting output files in description")
+    files = {}
+    if description['outFiles'] and description['outFiles'] != "NULL":
+        out_files = split(description["outFiles"])
+        length = len(out_files)
+        ddm_endpoint = split(description.get("ddmEndPointOut"), min_len=length)
+        destination_se = split(description.get("fileDestinationSE"), min_len=length)
+        dblock_token = split(description.get("dispatchDBlockTokenForOut"), min_len=length)
+        dblock_tokens = split(description.get("prodDBlockTokenForOut"), min_len=length)
+        datasets = split(description.get("realDatasets"), min_len=length)
+        dblocks = split(description.get("destinationDblock"), min_len=length)
+        destination_dblock_token = split(description.get("destinationDBlockToken"), min_len=length)
+        scope = split(description.get("scopeOut"), min_len=length, fill_last=True)
+
+        for i, f in enumerate(out_files):
+            if f is not None:
+                files[f] = {
+                    "ddm_endpoint": ddm_endpoint[i],
+                    "storage_element": destination_se[i],
+                    "dispatch_dblock_token": dblock_token[i],
+                    "destination_dblock_token": destination_dblock_token[i],
+                    "dblock_token": dblock_tokens[i],
+                    "dataset": datasets[i],
+                    "dblock": dblocks[i],
+                    "scope": scope[i]
+                }
+
+    return fix_log(description, files)
+
+
+def one_or_set(array):
+    if len(array) < 1:
+        return join(array)
+
+    zero = array[0]
+
+    for i in array:
+        if i != zero:
+            return join(array)
+
+    return stringify_weird(zero)
+
+
+class JobDescription(object):
+    __holder = None
+    __key_aliases = {
+        'PandaID': 'job_id',  # it is job id, not PanDA
+        'transformation': 'command',  # making it more convenient
+        'jobPars': 'command_parameters',  # -.-
+        'coreCount': 'cores_number',
```

Comment:
maybe change to number_of_cores, or stick to core_count (which is an established term)

Reply:
I think, number_of_cores in that case better. Core_count — slightly differs in meaning...